specs/CookieManagement.md

```diff
@@ -0,0 +1,158 @@
+
+# Description
+
+One can create a `CookieManager` off a WebView to `GetCookies` via a `CookieList` collection, `SetCookie`, `DeleteCookies`, and `ClearAllCookies` in a WebView environment.
+
+# Examples
+
+The following code snippet demonstrates how the cookie management APIs can be use:
+
+```c#
+CoreWebView2CookieManager _cookieManager;
+
+private void WebView_CoreWebView2Ready(object sender, EventArgs e)
+{
+    _cookieManager = webView.CoreWebView2.CreateCookieManager();
+}
+
+void SetCookieCmdExecuted(object target, ExecutedRoutedEventArgs e)
+{
+    CoreWebView2Cookie cookie = _cookieManager.CreateCookie("CookieName", "CookieValue", ".bing.com", "/");
+    cookie.SameSite = CoreWebView2CookieSameSiteKind.None;
+    _cookieManager.SetCookie(cookie);
+}
+
+async void GetCookiesCmdExecuted(object target, ExecutedRoutedEventArgs e)
+{
+    CoreWebView2CookieList cookieList = await _cookieManager.GetCookiesAsync("https://www.bing.com");
+    for (uint i = 0; i < cookieList.Size; ++i)
+    {
+        CoreWebView2Cookie cookie = cookieList.GetValueAtIndex(i);
+        Console.WriteLine(CookieAsString(cookie));
+    }
+
+}
+
+void ClearAllCookiesCmdExecuted(object target, ExecutedRoutedEventArgs e)
+{
+    _cookieManager.ClearAllCookies();
+}
+
+void DeleteCookiesCmdExecuted(object target, ExecutedRoutedEventArgs e)
+{
+    _cookieManager.DeleteCookies("CookieName", "https://www.bing.com", "", "");
+}
+```
+
+# API Notes
+
+See [API Details](#api-details) section below for API reference.
+
+# API Details
+
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    /// Kind of cookie SameSite status used in the CoreWebView2Cookie class.
+    /// These fields match those as specified in https://developer.mozilla.org/docs/Web/HTTP/Cookies#.
+    /// Learn more about SameSite cookies here: https://tools.ietf.org/html/draft-west-first-party-cookies-07
+    enum CoreWebView2CookieSameSiteKind
+    {
+        /// None SameSite type. No restrictions on cross-site requests.
+        None = 0,
+        /// Lax SameSite type. The cookie will be sent with "same-site" requests, and with "cross-site" top level navigation.
+        Lax = 1,
+        /// Strict SameSite type. The cookie will only be sent along with "same-site" requests.
+        Strict = 2,
+    };
+
+    runtimeclass CoreWebView2
+    {
+        /// Create a new cookie manager object.
+        CoreWebView2CookieManager CreateCookieManager();
+    }
+
+    /// Add or delete an CoreWebView2Cookie or CoreWebView2Cookies,
+    /// or view the cookies. The changes would apply to the context of the user profile.
+    /// That is, other WebViews under the same user profile could be affected.
+    runtimeclass CoreWebView2CookieManager
+    {
+        /// Create a cookie object with a specified name, value, domain, and path.

```

Comment:
It would be good to note in the documentation that this only creates a cookie object and it is not added to the cookie manager until you call Set/AddCookie